shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/resultset/ResultSetUtil.java

```diff
@@ -115,14 +115,14 @@ public static Object convertBigDecimalValue(final Object value, final boolean ne
     }
   
     private static BigDecimal adjustBigDecimalResult(final BigDecimal value, final boolean needScale, final int scale) {
-        if (needScale) {
-            try {
-                return value.setScale(scale);
-            } catch (final ArithmeticException ex) {
-                return value.setScale(scale, BigDecimal.ROUND_HALF_UP);
-            }
-        }
-        return value;
+        if (!needScale){
+            return value;
+        }
+        try {
+            return value.setScale(scale);
+        } catch (final ArithmeticException ex) {
+            return value.setScale(scale, BigDecimal.ROUND_HALF_UP);
+        }
```

Comment:
This is guard clauses, please keep the original codes

Reply:
Hi, the code has been rolled back, please review. 
But I think the modified statement conforms to the guard clauses, and the code is more concise.

Reply:
Thank you